backend/src/battles/battles.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Controller, Get, Param, HttpCode, InternalServerErrorException, NotFoundException, BadRequestException } from '@nestjs/common'
+import { BattlesService } from './battles.service'
+import { BattleResultResponseDto } from './dto/battleResult.dto'
+
+@Controller('battles')
+export class BattlesController {
+  constructor(private readonly battlesService: BattlesService) {}
+
+  @Get(':id/result')
+  @HttpCode(200)
+  getBattleResult(@Param('id') battleId: string): BattleResultResponseDto {
+    try {
+      return this.battlesService.getBattleResult(battleId)
+    } catch (error) {
+      if (error instanceof NotFoundException || error instanceof BadRequestException) {
+        throw error
+      }
```

Comment:
```suggestion
      if (error instanceof HttpException) {
        throw error
      }
```

Service에서 Exception이 추가될수록 해당 부분도 수정되니 유지보수성이 낮다고 생각이 듭니다. 
어차피 `HttpException`을 상속받으니, 이런 방식은 어떠신가요?

Reply:
수정했습니다! 👍